web/routes/util.js

```diff
@@ -0,0 +1,68 @@
+'use strict'
+const fs = require( 'fs');
+
+const readDirectory = ( dir) => {
+  return new Promise( (resolve, reject) => {
+    fs.readdir( dir, (err, res) => {
+      if( err){
+        reject( err);
+      } else {
+        resolve( res);
+      }
+    });
+  });
+};
+
+module.exports = {
+  getDirPicCount: async function( dir) {
+    let image_count = -1;
+    try {
+      const listing = await readDirectory( dir);
+      image_count = listing.length;
+    } catch( e) {
+      if( e.code === "ENOENT") {
+        console.log( "creating missing directory:", dir)
+        // create missing directory
+        try {
+          fs.mkdirSync( dir, 484);
```

Comment:
What is this magic number `484`? can you please put into a named constant?

Reply:
gah sorry, lost the comment in the refactoring, 484 == 0744 ... linter won't allow octal numbers :(